src/sqlfluff/dialects/dialect_mysql.py

```diff
@@ -314,6 +330,142 @@ class CreateTableStatementSegment(ansi.CreateTableStatementSegment):
     )
 
 
+class CreateUserStatementSegment(ansi.CreateUserStatementSegment):
+    """`CREATE USER` statement.
+
+    https://dev.mysql.com/doc/refman/8.0/en/create-user.html
+    """
+
+    _string_literal = OneOf(
+        Ref("QuotedLiteralSegment"), Ref("DoubleQuotedLiteralSegment")
+    )
+
+    _random_password = Sequence("RANDOM", "PASSWORD")
+    _auth_plugin = Ref("ObjectReferenceSegment")
+    _auth_string = _string_literal
+
+    _initial_auth_option = Sequence(
+        "INITIAL",
+        "AUTHENTICATION",
+        "IDENTIFIED",
+        OneOf(
+            Sequence("BY", OneOf(_random_password, _auth_string)),
+            Sequence(
+                "WITH",
+                _auth_plugin,
+                "AS",
+                _auth_string,
+            ),
+        ),
+    )
```

Comment:
This is only used in one place. So is it worth pulling out to it's own grammar?

Actually it's the same with ALL of the below.

Reply:
ibid https://github.com/sqlfluff/sqlfluff/pull/3289/commits/9e475bdb1e1683e9cf3bf150e103a1323df7a381